### PR TITLE
Add Sensirion SEN5X lib + I2C driver

### PR DESCRIPTION
**Pull request by tyeth**
## Description:
Adds SEN5x series of Particulate sensors (1-10PPM/Temp/Humidity/NOx/VOC)
 also SEN54 (voc+ppm+temp+humidity) or SEN50 (PPM only)
see https://sensirion.com/products/catalog/SEN55/ etc

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [-] The code change is tested and works with Tasmota core ESP8266 V.2.7.4.9
  - [x] The code change is tested and works with Tasmota core ESP32 V.2.0.6
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

**Comment by tyeth** [comment]
Original commits: https://github.com/arendst/Tasmota/compare/development...tyeth:Tasmota:add-sen5x

**Comment by Jason2866** [comment]
Where is the Tasmota driver? This PR provides only the Sensirion lib.

**Comment by tyeth** [comment]
Sorry, my mistake, I squashed it out of existence. Will fix when I get home

File: tasmota/tasmota_xsns_sensor/xsns_103_sen5x.ino [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+float sen5x_abshum;
+float sen5x_massConcentrationPm1p0 = NAN;
+float sen5x_massConcentrationPm2p5 = NAN;
+float sen5x_massConcentrationPm4p0 = NAN;
+float sen5x_massConcentrationPm10p0 = NAN;
+float sen5x_ambientHumidity = NAN;
+float sen5x_ambientTemperature = NAN;
+float sen5x_vocIndex = NAN;
+float sen5x_noxIndex = NAN;
+
```
Comment (barbudor) [inline tasmota/tasmota_xsns_sensor/xsns_103_sen5x.ino:51]: That's too many globals that are lost even if the driver is not in use
Please move to a dynamically allocated structure which will use heap only when the driver is enabled and the device detected
See recent drivers for the structure

Comment (tyeth) [reply tasmota/tasmota_xsns_sensor/xsns_103_sen5x.ino]: I've gone for one of the more recent implementations. I'm a bit unsure of how best to do it, but does this achieve the job? https://github.com/tyeth/Tasmota/commit/58687573c81d57cac5756c2faa29f571c8cf2770

Comment (barbudor) [reply tasmota/tasmota_xsns_sensor/xsns_103_sen5x.ino]: Exactly
Well done
May thanks

**Comment by tyeth** [comment]
After loading a copy via tasmotaCompiler, with the air sensors enabled, it's showing as an SPS30, so something needs to change. Fail.

**Comment by sfromis** [comment]
I2C address clashes leading to faulty detection is not too unusual, and it is commonly recommended for the user to mitigate such issues by using the command `I2CDriver` to deactivate one of them. As the sensors cannot be used at the same time anyway, it is not required to mitigate the issue in the other driver, but if it is easy to do....

**Comment by Jason2866** [comment]
@tyeth As sfromis explained the current driver regarding i2c handling is okay.
You can remove the Draft status and it is good to be merged.
Enhancing and refactoring other drivers should be done in an extra PR.

**Comment by tyeth** [comment]
I've made significant improvements, especially around logging, more thorough testing in situ, and error handling depending on sensor model (no VOC/NOx/Temp/Hum with SEN50 - only particules, SEN54 missing only VOC, sen55 full features). It's also been thoroughly tested on both I2C busses with other sensors attached.

 I must confess my sins, it was poorly tested and actually broken when first submitted so lessons learned and my apologies. I had problems building firmwares with all needed features, using tasmocompiler seemed to be okay, but my custom builds always showed max or min values for all sensors. I was trying to do too much at once as most newbies do - requiring display/lvgl/custom-defines, especially stumbled with certain unexpected defines like the AHT1x when I'm using the AHT20, they conflict and the docs says AHT1x isn't included in default builds but it is...

Happy enough now once the SPS30 pull request is merged. They can be merged separately but would be grateful if they made it into the next release together.

**Comment by tyeth** [comment]
Sorry, forgot there were missing units. Will fix immediately.

**Comment by Jason2866** [comment]
Please remove the sps30 driver changes from this PR. The SPS30 PR will be merged together or before this PR ;-)

**Comment by tyeth** [comment]
Ah sorry, slipped back in with the units ;)

**Comment by tyeth** [comment]
Should be all good, retested (ignore double percent in screenshot as it's fixed) and checked for spurious commits :)
image

**Comment by tyeth** [comment]
Seen the failure, I'll add a guard around wire1 for esp32, falling back to wire if necessary.

**Pull request merged**